packages/button/Button/IconButton.tsx

```diff
@@ -1,5 +1,27 @@
-import * as React from "react"
-import GenericButton, { IconButtonProps } from "./components/GenericButton"
+import React, { ComponentType } from "react"
+import GenericButton, {
+  GenericProps,
+  BadgeProps,
+  CustomButtonProps,
+  WorkingProps,
+  WorkingUndefinedProps,
+} from "./components/GenericButton"
+
+type IconButtonProps = GenericProps &
+  (WorkingProps | WorkingUndefinedProps) & {
+    label: string
+    primary?: boolean
+    destructive?: boolean
+    secondary?: boolean
+    form?: boolean
+    badge?: BadgeProps
+    type?: "submit" | "reset" | "button"
+    fullWidth?: boolean
+    component?: ComponentType<CustomButtonProps>
+    iconPosition?: "start" | "end"
```

Comment:
I _think_ `iconPosition` isn't a prop for `IconButton` currently.

Reply:
You're right doug, I must have got my typings mixed up when refactoring LabelPropsGeneric.